engine/src/main/java/org/destinationsol/game/screens/MapScreen.java

```diff
@@ -76,5 +101,94 @@ public void updateCustom(SolApplication solApplication, SolInputManager.InputPoi
+                        canCreate = false;
+                        break;
+                    }
+                }
+
+                if (canCreate) {
+                    disableWaypointButtons(true);
+                    WaypointCreationScreen waypointCreationScreen = game.getScreens().waypointCreationScreen;
+                    waypointCreationScreen.setWaypointPos(worldPosition);
+
+                    solApplication.getInputManager().setScreen(solApplication, game.getScreens().mapScreen);
+                    solApplication.getInputManager().addScreen(solApplication, waypointCreationScreen);
+                }
+                addWaypointControl.setDisplayName(NEW_WAYPOINT_TEXT);
+                isPickingWaypointSpot = false;
+            }
+        }
+
+        if (removeWaypointControl.isJustOff() && !isPickingWaypointToRemove) {
+            isPickingWaypointToRemove = true;
+            removeWaypointControl.setDisplayName(CANCEL_TEXT);
+            addWaypointControl.setEnabled(false);
+        } else if (removeWaypointControl.isJustOff() && isPickingWaypointToRemove) {
+            removeWaypointControl.setDisplayName(REMOVE_WAYPOINT_TEXT);
+            addWaypointControl.setEnabled(true);
+            isPickingWaypointToRemove = false;
+        }
+        else if (isPickingWaypointToRemove) {
+            if (inputPointers[0].isJustUnPressed()) {
+                Vector2 clickPosition = new Vector2(inputPointers[0].x, inputPointers[0].y);
+                Vector2 realPosition = screenPositionToWorld(clickPosition, game.getCam().getPosition(), mapZoom);
+
+                ArrayList<Waypoint> waypoints = game.getHero().getWaypoints();
+                for (int w = 0; w < waypoints.size(); w++) {
+                    Waypoint waypoint = waypoints.get(w);
+                    if (waypoint.position.x > realPosition.x-MIN_WAYPOINT_DIST && waypoint.position.x < realPosition.x+MIN_WAYPOINT_DIST &&
+                        waypoint.position.y > realPosition.y-MIN_WAYPOINT_DIST && waypoint.position.y < realPosition.y+MIN_WAYPOINT_DIST) {
+                        game.getHero().removeWaypoint(waypoint);
+                        game.getObjectManager().removeObjDelayed(waypoint);
+                    }
+                }
+                isPickingWaypointToRemove = false;
+                addWaypointControl.setEnabled(true);
+            }
+        }
+    }
+
+    public Vector2 screenPositionToWorld(Vector2 screenPos, Vector2 camPos, float mapZoom) {
+        float ratio = (float)Gdx.graphics.getWidth()/(float)Gdx.graphics.getHeight();
+        screenPos.scl(5);
+        screenPos.scl(mapZoom);
+
+        Vector2 finalPosition = new Vector2(camPos);
+        finalPosition.add(screenPos);
+
+        finalPosition.x -= (ratio * mapZoom)/(2.f/5);
+        finalPosition.y -= (mapZoom)/(2.f/5);
+        return finalPosition;
+    }
+
+    public void disableWaypointButtons(boolean value) {
```

Comment:
This function can be simplied by removing the if-Statement and then changing the calls to `setEnabled(!value)`.
(If there is a setDisabled() function, use it instead.)